src/cart-icon/index.ts

```diff
@@ -0,0 +1,38 @@
+import { alias, tag, Events, Selectors, Tag } from '@storefront/core';
+
+@alias('quantity')
+@tag('gb-cart-icon', require('./index.html'), require('./index.css'))
+class CartIcon {
+  state: CartIcon.State = {
+    quantity: 0
+  };
+
+  constructor() {
+    const cart = this.select(Selectors.cart);
+    this.state = { ...this.state, quantity: cart.content.quantity };
+  }
+
+  init() {
+    this.flux.on(Events.CART_UPDATED, this.updateQuantity);
+  }
+
+  updateQuantity = () => {
+    const cart = this.select(Selectors.cart);
+    this.set({ quantity: cart.content.quantity });
```

Comment:
How about const quantity = this.select(Selectors.cart).content.quantity

Then just this.set({ quantity })